Formula/grafana-cloud-agent.rb

```diff
@@ -0,0 +1,103 @@
+class GrafanaCloudAgent < Formula
+  desc "Lightweight subset of Prometheus and more, optimized for Grafana Cloud"
+  homepage "https://grafana.com/products/cloud/"
+  url "https://github.com/grafana/agent/archive/v0.8.0.tar.gz"
+  sha256 "3b8cb72793888f676fb205ced065da173429289dd29c77b79b095350ec28bf35"
+  license "Apache-2.0"
+
+  livecheck do
+    url "https://github.com/grafana/agent/releases/latest"
+    regex(%r{href=.*?/tag/v?(\d+(?:\.\d+)+)["' >]}i)
+  end
+
+  depends_on "go" => :build
+
+  def install
+    mkdir_p buildpath/"src/github.com/grafana"
+    ln_sf buildpath, buildpath/"src/github.com/grafana/agent"
+
+    system "make", "agent"
+
+    bin.install "./cmd/agent/agent"
+    mv "#{bin}/agent", "#{bin}/grafana-cloud-agent"
+
+    (bin/"grafana-cloud-agent_brew_services").write <<~EOS
+      #!/bin/bash
+      exec #{bin}/grafana-cloud-agent $(<#{etc}/grafana-cloud-agent.args)
+    EOS
+
+    (buildpath/"grafana-cloud-agent.args").write <<~EOS
+      --config.file #{etc}/grafana-cloud-agent.yml
+    EOS
+
+    (buildpath/"grafana-cloud-agent.yml").write <<~EOS
+      server:
+        log_level: info
+        http_listen_port: 12345
+
+      prometheus:
+        wal_directory: /tmp/agent
+        global:
+          scrape_interval: 15s
+
+      integrations:
+        node_exporter:
+          enabled: true
+          rootfs_path: /host/root
+          sysfs_path: /host/sys
+          procfs_path: /host/proc
+        prometheus_remote_write:
+          - url: https://prometheus-us-central1.grafana.net/api/prom/push
+            basic_auth:
+              username: USERNAME
+              password: PASSWORD
+    EOS
```

Comment:
This should come from upstream

Reply:
Do you mean that the upstream should contain a default configuration file? To author this formula I followed the [Prometheus formula](https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/prometheus.rb) as this agent has a very similar use case. The agent purposefully does not come with a default configuration file as it can be installed and configured in many different ways and for many different use cases. For Mac users, however, I thought that a good default that monitors the basic machine out of the box would be a better UX for end-users.

Reply:
If upstream doesn't ship a default Homebrew shouldn't either.